### PR TITLE
feat: 🎸 new preselectedAgency logic, styles and texts

### PR DESCRIPTION
**Pull request by koepferd**
## Proposed Changes

  - refactored `<SelectedAgencyInfo />` to `<PreselectedAgency />`
  - implemented new display logic
  - new styles
  - new texts for `<AgencySelection />` in case 'preselectedAgency'

### CODE REVIEW

File: src/components/agencySelection/AgencyInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+					{displayAgencyInfo.teamAgency && (
+						<div className="agencyInfo__teamAgency">
+							<InfoIcon />
+							{translate(
+								'registration.agency.preselected.isTeam'
+							)}
+						</div>
+					)}
+					{displayAgencyInfo.name && (
+						<div className="agencyInfo__name">
```
Comment (iam-robin) [inline src/components/agencySelection/AgencyInfo.tsx]: is this a general text style from the design system? If so, can we use the headline/text component here?

File: src/components/agencySelection/AgencyInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+							{displayAgencyInfo.name}
+						</div>
+					)}
+					{displayAgencyInfo.description && (
+						<div
```
Comment (iam-robin) [inline src/components/agencySelection/AgencyInfo.tsx]: same here, text component possible?

File: src/components/agencySelection/AgencySelection.tsx [hunk 3]
```diff
@@ -36,21 +30,19 @@ export interface AgencySelectionProps {
...
 	const [selectedPostcode, setSelectedPostcode] = useState('');
 	const [selectedAgencyId, setSelectedAgencyId] = useState<
 		number | undefined
 	>(undefined);
 	const [autoSelectAgency, setAutoSelectAgency] = useState(false);
 	const [autoSelectPostcode, setAutoSelectPostcode] = useState(false);
-	const [displayAgencyInfo, setDisplayAgencyInfo] = useState<
-		AgencyDataInterface
-	>(null);
+	const [preselectedAgency, setPreselectedAgency] = useState(
```
Comment (iam-robin) [inline src/components/agencySelection/AgencySelection.tsx]: can you type state with the correct data type?

File: src/components/agencySelection/AgencySelection.tsx [hunk 4]
```diff
@@ -208,21 +200,39 @@ export const AgencySelection = (props: AgencySelectionProps) => {
 				<>
 					{props.isProfileView && (
 						<h5>
```
Comment (iam-robin) [inline src/components/agencySelection/AgencySelection.tsx]: please use headline component here

File: src/components/agencySelection/AgencySelection.tsx [hunk 5]
```diff
@@ -184,6 +185,13 @@ export const AgencySelection = (props: AgencySelectionProps) => {
 				'registration.agencySelection.intro.point2',
 				'registration.agencySelection.intro.point3'
 		  ];
+	const getHeadlineForProfileView = () => {
```
Comment (iam-robin) [inline src/components/agencySelection/AgencySelection.tsx]: can the condition be done inside the text property of the headline component?

Comment (koepferd) [reply src/components/agencySelection/AgencySelection.tsx]: yes, but I wanted it to be more readable, but for consistency I will move it to the headline component. thanks

**Review by iam-robin — DISMISSED** [review]

**Review by iam-robin — APPROVED** [review]

**Pull request merged**